client/my-sites/earn/memberships/test/coupons-list.jsx

```diff
@@ -0,0 +1,235 @@
+	discount_type: COUPON_DISCOUNT_TYPE_PERCENTAGE,
+	discount_value: 0,
+	discount_percentage: 90,
+	discount_currency: 'USD',
+};
+const testCoupon2 = {
+	ID: 2,
+	coupon_code: 'COUPON2',
+	cannot_be_combined: false,
+	can_be_combined: true, // TODO: remove after backend migration to 'cannot_be_combined';
+	first_time_purchase_only: false,
+	start_date: '2023-12-25',
+	end_date: '2024-10-23',
+	plan_ids_allow_list: [],
+	limit_per_user: 0,
+	use_duration: false,
+	use_email_allow_list: true,
+	duration: COUPON_DURATION_FOREVER,
+	email_allow_list: [ '*@*.edu', '*@*.org' ],
+	discount_type: COUPON_DISCOUNT_TYPE_PERCENTAGE,
+	discount_value: 0,
+	discount_percentage: 90,
+	discount_currency: 'USD',
+};
+const testCoupon3 = {
+	ID: 3,
+	coupon_code: 'COUPON3',
+	cannot_be_combined: true,
+	can_be_combined: false, // TODO: remove after backend migration to 'cannot_be_combined';
+	first_time_purchase_only: true,
+	start_date: '2023-12-25',
+	end_date: '2024-10-23',
+	plan_ids_allow_list: [],
+	limit_per_user: 0,
+	use_duration: true,
+	use_email_allow_list: true,
+	duration: COUPON_DURATION_3_MONTHS,
+	email_allow_list: [],
+	discount_type: COUPON_DISCOUNT_TYPE_AMOUNT,
+	discount_value: 10,
+	discount_percentage: 0,
+	discount_currency: 'USD',
+};
+
+const initialState = {
+	sites: { items: { 1: { ID: 1 } }, features: { 1: { data: { active: [ 'donations' ] } } } },
+	ui: { selectedSiteId: 1 },
+	memberships: {
+		couponList: {
+			items: {},
+		},
+		productList: {
+			items: {},
+		},
+		settings: {},
+	},
+};
+
+function WrappedCouponsList( props ) {
+	const testEnabled = props.testEnabled;
+	const testCoupons = props.testCoupons;
+	const mockStore = configureStore();
+	const mockState = {
+		...initialState,
+		memberships: {
+			settings: {
+				1: {
+					couponsAndGiftsEnabled: testEnabled,
+					connectedAccountId: 123,
+					connectedAccountMinimumCurrency: {
+						USD: 0.5,
+					},
+				},
+			},
+			couponList: {
+				items: {
+					1: testCoupons,
+				},
+			},
+			productList: {
+				items: {
+					1: [ testProduct1, testProduct2 ],
+				},
+			},
+		},
+	};
+	const store = mockStore( mockState );
+
+	return (
+		<Provider store={ store }>
+			<CouponsList />
+		</Provider>
+	);
+}
+
+describe( 'CouponsList', () => {
+	let modalRoot;
+
+	beforeEach( () => {
+		jest.clearAllMocks();
+
+		useTranslate.mockImplementation( mockUseTranslate );
+		translate.mockImplementation( mockTranslate );
+		__.mockImplementation( mockTranslate );
+
+		modalRoot = document.createElement( 'div' );
+		modalRoot.setAttribute( 'id', 'wpcom' );
+		document.body.appendChild( modalRoot );
+		Modal.setAppElement( modalRoot );
+	} );
+
+	afterEach( () => {
+		unmountComponentAtNode( modalRoot );
+		document.body.removeChild( modalRoot );
+		modalRoot = null;
+		global.Date = RealDate;
+	} );
+
+	test( 'should render nothing when feature is disabled', () => {
+		const { container } = render( <WrappedCouponsList /> );
+		expect( container ).toMatchSnapshot();
```

Comment:
Let's avoid using snapshots - they add little meaning in our tests. Let's instead assert that a certain element is empty.